plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -293,4 +302,22 @@ public long getRetainedSizeInBytes()
             return INSTANCE_SIZE + sizeOf(files, value -> estimatedSizeOf(value, TrinoFileStatus::getRetainedSizeInBytes));
         }
     }
+
+    private record CacheKey(Location location, SchemaTableName schemaTableName)
+    {
+        private static final long INSTANCE_SIZE = instanceSize(CacheKey.class);
+
+        private CacheKey(Location location, SchemaTableName schemaTableName)
+        {
+            this.location = requireNonNull(location, "location is null");
+            this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
+        }
+
+        public long getRetainedSizeInBytes()
+        {
+            return INSTANCE_SIZE +
+                    estimatedSizeOf(location.toString()) +
+                    (schemaTableName != null ? estimatedSizeOf(schemaTableName.toString()) : 0);
```

Comment:
`schemaTableName` is never null as enforced by the constructor validation, and you can use `schemaTableName.getRetainedSizeInBytes()`

Reply:
done! good call!